elements/pl-checkbox/pl-checkbox.py

```diff
@@ -148,19 +150,26 @@ def render(element_html, data):
 
     correct_answer_list = data['correct_answers'].get(name, [])
     correct_keys = [answer['key'] for answer in correct_answer_list]
+    # Pointer used to index feedback
+    ptr = 0
```

Comment:
I believe this and the below logic is more complex than it needs to be. I don't think we need to conditionally increment this - something like this is simpler to understand and more Pythonic:

```py
for i, answer in display_answers:
	answer_html = {
    # ...
    'feedback': feedback[i] if feedback and answer['key'] in submitted_keys else None
  }
```

Apologies if there's a specific reason that we're conditionally incrementing the index and I missed it. If there is, let's document it.

Reply:
The conditional is used to make sure we don't go over a piece of feedback as we're iterating through the answer choices.

For example, let's say I have five questions in `display_answers` and I select two. `feedback` is now two elements long.

As I'm iterating through `display_answers`, `i` might not index to the correct item in `feedback`.

After some thought, I can get rid of the pointer syntax by making `feedback` a dictionary, or a list of dictionaries. Let me know your thoughts.

Reply:
Ahhh, I understand - the elements of `feedback` correspond to what's been selected, not necessarily the elements of `display_answers`?

I think there are two potential ways to improve this:

1. refactor this so that `feedback` is a dict that maps answer keys to their feedback:
```py
{
  'a': 'a feedback',
  'b': 'b feedback',
  'c': None
}
```
2. refactor this so that feedback is an array whose elements correspond 1:1 with the items in `display_answers`:
```py
[
  'a feedback',
  'b feedback',
  None
]
```

In both cases, `feedback` would contain feedback for *all* answer choices, not just the ones that the user chose for a particular submission. IMO, that makes this easier to reason about. Whether or not we actually end up rendering the feedback would still be determined by if the user chose that coice, etc.

Reply:
Got it. Will push changes shortly.